tests/test_api.py

```diff
@@ -211,3 +211,22 @@ def test_warnings():
 def test_commutativity():
     o = toml.loads(toml.dumps(TEST_DICT))
     assert o == toml.loads(toml.dumps(o))
+
+
+def test_comment_preserve_decoder_encoder():
+    test_str = """[[products]]
+name = "Nail"
+sku = 284758393
+num = [ 1, #Number 1
+        2, # Number 2
+        3,
+      ]
+# This is a comment
+color = "gray" # Hello World
+"""
```

Comment:
I think it would be good to test scenarios where comments are valid toml.

e.g. a commented out key with inline table, commented out key with multiline string, commented out key with array